source/base/mpi.cc

```diff
@@ -498,60 +437,9 @@ namespace Utilities
       const MPI_Comm &                 mpi_comm,
       const std::vector<unsigned int> &destinations)
     {
-      const unsigned int n_procs = Utilities::MPI::n_mpi_processes(mpi_comm);
-
-      for (const unsigned int destination : destinations)
-        {
-          (void)destination;
-          AssertIndexRange(destination, n_procs);
-          Assert(destination != Utilities::MPI::this_mpi_process(mpi_comm),
-                 ExcMessage(
-                   "There is no point in communicating with ourselves."));
-        }
-
-      // Calculate the number of messages to send to each process
-      std::vector<unsigned int> dest_vector(n_procs);
-      for (const auto &el : destinations)
-        ++dest_vector[el];
-
-#  if DEAL_II_MPI_VERSION_GTE(2, 2)
-      // Find out how many processes will send to this one
-      // MPI_Reduce_scatter(_block) does exactly this
-      unsigned int n_recv_from = 0;
-
-      const int ierr = MPI_Reduce_scatter_block(
-        dest_vector.data(), &n_recv_from, 1, MPI_UNSIGNED, MPI_SUM, mpi_comm);
-
-      AssertThrowMPI(ierr);
-
-      return n_recv_from;
-#  else
-      // Find out how many processes will send to this one
-      // by reducing with sum and then scattering the
-      // results over all processes
-      std::vector<unsigned int> buffer(dest_vector.size());
-      unsigned int              n_recv_from = 0;
-
-      int ierr = MPI_Reduce(dest_vector.data(),
-                            buffer.data(),
-                            dest_vector.size(),
-                            MPI_UNSIGNED,
-                            MPI_SUM,
-                            0,
-                            mpi_comm);
-      AssertThrowMPI(ierr);
-      ierr = MPI_Scatter(buffer.data(),
-                         1,
-                         MPI_UNSIGNED,
-                         &n_recv_from,
-                         1,
-                         MPI_UNSIGNED,
-                         0,
-                         mpi_comm);
-      AssertThrowMPI(ierr);
-
-      return n_recv_from;
-#  endif
+      return compute_point_to_point_communication_pattern(mpi_comm,
+                                                          destinations)
+        .size();
```

Comment:
I don't see a point in keeping the code above. Calling the other function per default shouldn't be a problem in particular since in the normal case the NBX path is taken.